### ISSUE TITLE
C++ Exception trainer.TrainMiniBatch

### ISSUE THREAD

**Issue by chrubiofa** [Microsoft/CNTK#1181]
Hi,

I am running into an exception which is not quite clear.  
I have included my source code and sample training data, I am sure someone can point me in the correct direction. 
For starters, I am using the latest release running Visual Studio 2013 Update 5,  
I include the common.h and common.cpp files from the tests (version 2.0) directory and created a simple
void main() calling the attached sample code. Including test data that is called as well.

[source.txt](https://github.com/Microsoft/CNTK/files/661867/source.txt)
[test.txt](https://github.com/Microsoft/CNTK/files/661864/test.txt)

I get the following exception: 
**DataFor: FrameRange's dynamic axis is inconsistent with matrix: {numTimeSteps:18
3, numParallelSequences:2, sequences:[{seqId:0, s:0, begin:0, end:148}, {seqId:1
, s:1, begin:0, end:183}, {seqId:18446744073709551615, s:0, begin:148, end:183}]
} vs. {numTimeSteps:1, numParallelSequences:2, sequences:[{seqId:0, s:0, begin:0
, end:1}, {seqId:1, s:1, begin:0, end:1}]}**

What I want is an identical C++ implementation of the brain script network described like the following;
```
        BiRecurrentLSTMLayer {outDim} = {
            F = RecurrentLSTMLayer {outDim, goBackwards=false}
            G = RecurrentLSTMLayer {outDim, goBackwards=true}
            apply (x) = Splice (F(x):G(x))
        }.apply
        model = Sequential (
            BiRecurrentLSTMLayer {hiddenDim} :
	    BatchNormalizationLayer {} :
            DenseLayer {4}
        )
```

Thank you and warm regards
Chris.

**Comment by raaaar** [comment]
Hello. Looks like your network definition is inconsistent with your data. Here you explicitly say that labels only have a single dynamic axis (i.e, only batch axis, no sequence axis):

```
auto labels = InputVariable({ numOutputClasses }, true, DataType::Float, labelsName, { Axis::DefaultBatchAxis() }); 
```
 So, the network expects a single label per sequence. Yet in the input data, each sample in a sequence has a corresponding label:

```
11	|WV 0.269459 ... 0.358702	|O 2:1	|#xxx
11	|WV 0.313503 ... -0.058333	|O 2:1	|#xxx
11	|WV 0.055909 ... 0.138958	|O 3:1	|#xxx
11	|WV 0.078846 ... 0.092946	|O 0:1	|#xxx

```

[This python definition](https://github.com/Microsoft/CNTK/blob/fseide/slupy/Tests/EndToEndTests/CNTKv2Python/Examples/language_understanding_test.py#L70) is almost identical to the bs snippet above. It should be fairly straight forward to translate it to C++, but if you run into any issues there, maybe @frankseide or @n17s would be able to help.

**Comment by chrubiofa** [comment]
That was the issue, one of the functions was returning a different axis then expected.
Thanks,

Regards.

**Comment by anguoyang** [comment]
i have the same problem, but there is no problem when using the tensorflow back-end  in keras, what is the  problem with cntk?
